src/urllib3/request.py

```diff
@@ -67,6 +71,11 @@ def request(self, method, url, fields=None, headers=None, **urlopen_kw):
         :meth:`request_encode_url`, :meth:`request_encode_body`,
         or even the lowest level :meth:`urlopen`.
         """
+        if headers is None:
```

Comment:
Instead of adding the value upstream we should add it within `HTTPConnection` so we aren't modifying any upstream behaviors.

Reply:
Done in the original work.